### ISSUE TITLE
Feature Request: Perspective Transforms

### ISSUE THREAD

**Issue by bcarpenter-pub** [tensorflow/tensorflow#5632]
Currently image processing has some image manipulation functions like brightness adjustment or contrast adjustment.  It would be quite useful to have perspective transforms or other affine transformations to help distort training data.

**Comment by Nessphoro** [comment]
Hmm, I wonder if you could use the spatial transformer code to achieve this.
E.g. generate a sampling grid [0, height] x [0, width], apply your transformation (via batch_matmul) and then collect them exactly as in the spatial_transformer code.

**Comment by girving** [comment]
This is a reasonable feature request, but we may not get to it in the near term.  Contributions welcome!

**Comment by artcg** [comment]
I will work on this

**Comment by astromme** [comment]
Can this be done with https://www.tensorflow.org/api_docs/python/tf/contrib/image/transform?

**Comment by aselle** [comment]
@astromme, yes, this already is supported there. Closing as this feature already exists.

**Comment by Plutoberth** [comment]
I'm sorry about opening this so long after it was closed, but although this feature is implemented with tf.contrib.image.transform, that function does not allow gradients to backpropagate through it. So I'd call it a partial implementation.
